src/Razor/Microsoft.AspNetCore.Razor.Language/src/Syntax/InternalSyntax/SyntaxTokenCache.cs

```diff
@@ -60,7 +64,7 @@ public static SyntaxToken GetCachedToken(SyntaxKind kind, string content)
             var idx = indexableHash & CacheMask;
             var e = s_cache[idx];
 
-            if (e.Hash == hash && e.Token.Kind == kind && e.Token.Content == content)
+            if (e.Hash == hash && e.Token != null && e.Token.Kind == kind && e.Token.Content == content)
```

Comment:
```suggestion
            var isValidToken = e.Token != null && e.Token.Kind == kind && e.Token.Content == content;
            if (e.Hash == hash && isValidToken)
```

Reply:
Could I do this in the main? Since this is a port, I'd prefer having this be identical to the 6.0 one.

Reply:
Yep -- not blocking on it. Just figured it makes this easier to understand moving forward.